diff -urxN Master0/texmf-dist/scripts/texlive/updmap.pl Master1/texmf-dist/scripts/texlive/updmap.pl
--- Master0/texmf-dist/scripts/texlive/updmap.pl	2016-11-23 12:46:26.000000000 +0900
+++ Master1/texmf-dist/scripts/texlive/updmap.pl	2016-12-11 18:35:59.000000000 +0900
@@ -144,6 +144,18 @@
     type     => "any",
     default  => "",
   },
+  scEmbed            => {
+    type     => "any",
+    default  => "noEmbed",
+  },
+  tcEmbed            => {
+    type     => "any",
+    default  => "noEmbed",
+  },
+  koEmbed            => {
+    type     => "any",
+    default  => "noEmbed",
+  },
 );
 
 &main();
@@ -997,6 +1009,9 @@
   my ($pxdviUse, $pxdviUse_origin) = get_cfg('pxdviUse');
   my ($kanjiEmbed, $kanjiEmbed_origin) = get_cfg('kanjiEmbed');
   my ($kanjiVariant, $kanjiVariant_origin) = get_cfg('kanjiVariant');
+  my ($scEmbed, $scEmbed_origin) = get_cfg('scEmbed');
+  my ($tcEmbed, $tcEmbed_origin) = get_cfg('tcEmbed');
+  my ($koEmbed, $koEmbed_origin) = get_cfg('koEmbed');
 
   # pxdvi is optional, and off by default.  Don't create the output
   # directory unless we are going to put something there.
@@ -1018,6 +1033,12 @@
          .      "$kanjiEmbed ($kanjiEmbed_origin)"
          . "\n  kanjiVariant replacement string  : "
          .      "$kanjiVariant ($kanjiVariant_origin)"
+         . "\n  scEmbed replacement string    : "
+         .      "$scEmbed ($scEmbed_origin)"
+         . "\n  tcEmbed replacement string    : "
+         .      "$tcEmbed ($tcEmbed_origin)"
+         . "\n  koEmbed replacement string    : "
+         .      "$koEmbed ($koEmbed_origin)"
          . "\n  create a mapfile for pxdvi       : "
          .      "$pxdviUse ($pxdviUse_origin)"
          . "\n\n");
@@ -1864,6 +1885,9 @@
   #
   my ($kanjiEmbed, $kanjiEmbed_origin) = get_cfg('kanjiEmbed');
   my ($kanjiVariant, $kanjiVariant_origin) = get_cfg('kanjiVariant');
+  my ($scEmbed, $scEmbed_origin) = get_cfg('scEmbed');
+  my ($tcEmbed, $tcEmbed_origin) = get_cfg('tcEmbed');
+  my ($koEmbed, $koEmbed_origin) = get_cfg('koEmbed');
   #
   # go through all map files and check that the text is properly replaced
   # after the replacement check that the generated map file actually
@@ -1871,10 +1895,15 @@
   #
   for my $l (@l) {
     for my $m (keys %{$alldata->{'updmap'}{$l}{'maps'}}) {
-      if ($m =~ m/\@kanjiEmbed@/ || $m =~ m/\@kanjiVariant@/) {
-        my $newm = $m;
-        $newm =~ s/\@kanjiEmbed@/$kanjiEmbed/;
-        $newm =~ s/\@kanjiVariant@/$kanjiVariant/;
+      my $newm = $m;
+      # do all kinds of substitutions
+      $newm =~ s/\@kanjiEmbed@/$kanjiEmbed/;
+      $newm =~ s/\@kanjiVariant@/$kanjiVariant/;
+      $newm =~ s/\@scEmbed@/$scEmbed/;
+      $newm =~ s/\@tcEmbed@/$tcEmbed/;
+      $newm =~ s/\@koEmbed@/$koEmbed/;
+      if ($newm ne $m) {
+        # something was substituted
         if (locateMap($newm)) {
           # now we have to update various linked items
           $alldata->{'updmap'}{$l}{'maps'}{$newm}{'type'} =
@@ -2292,6 +2321,9 @@
     Whether maps for pxdvi (Japanese-patched xdvi) are under updmap's control.
   kanjiEmbed            (any string)
   kanjiVariant          (any string)
+  scEmbed               (any string)
+  tcEmbed               (any string)
+  koEmbed               (any string)
     See below.
   LW35                  URWkb,URW,ADOBEkb,ADOBE  (default URWkb)
     Adapt the font and file names of the standard 35 PostScript fonts.
@@ -2311,7 +2343,9 @@
   in the map lines.  If a map contains the string \@kanjiEmbed\@, then
   this will be replaced by the value of that option; similarly for
   kanjiVariant.  In this way, users of Japanese TeX can select different
-  fonts to be included in the final output.
+  fonts to be included in the final output.  The counterpart for
+  Simplified Chinese, Traditional Chinese and Korean fonts are
+  scEmbed, tcEmbed and koEmbed respectively.
 
 Explanation of trees and files normally used:
 
diff -urxN Master0/texmf-dist/web2c/updmap-hdr.cfg Master1/texmf-dist/web2c/updmap-hdr.cfg
--- Master0/texmf-dist/web2c/updmap-hdr.cfg	2016-05-11 12:34:15.000000000 +0900
+++ Master1/texmf-dist/web2c/updmap-hdr.cfg	2016-12-11 18:40:58.000000000 +0900
@@ -85,7 +85,7 @@
 dvipdfmDownloadBase14 true
 
 
-# kanjiEmbed
+# kanjiEmbed, scEmbed, tcEmbed, koEmbed
 #
 # updmap generates a file kanjix.map for usage with dvipdfmx. With this
 # option one can switch between not embedding any kanji font, or
@@ -95,7 +95,14 @@
 # @kanjiEmbed@ in a map file name will be replaced with this value, and
 # the respective map file used.
 #
+# @kanjiEmbed@ is for Japanese (jfontmaps); the counterpart
+# for Simplified Chinese, Traditional Chinese and Korean
+# are @scEmbed@, @tcEmbed, @koEmbed@ respectively.
+#
 kanjiEmbed ipaex
+scEmbed arphic
+tcEmbed arphic
+koEmbed baekmuk
 
 
 #  (end of updmap-hdr.cfg)
