diff -urxN Master0/tlpkg/tlpsrc/japanese-otf.tlpsrc Master1/tlpkg/tlpsrc/japanese-otf.tlpsrc
--- Master0/tlpkg/tlpsrc/japanese-otf.tlpsrc	2012-04-13 09:26:34.000000000 +0900
+++ Master1/tlpkg/tlpsrc/japanese-otf.tlpsrc	2016-12-11 18:29:40.000000000 +0900
@@ -1,2 +1,4 @@
-execute addKanjiMap otf-cktx.map
 execute addKanjiMap otf-@kanjiEmbed@.map
+execute addKanjiMap otf-sc-@scEmbed@.map
+execute addKanjiMap otf-tc-@tcEmbed@.map
+execute addKanjiMap otf-ko-@koEmbed@.map
diff -urxN Master0/tlpkg/tlpsrc/uptex.tlpsrc Master1/tlpkg/tlpsrc/uptex.tlpsrc
--- Master0/tlpkg/tlpsrc/uptex.tlpsrc	2016-07-01 03:16:10.000000000 +0900
+++ Master1/tlpkg/tlpsrc/uptex.tlpsrc	2016-12-11 18:28:36.000000000 +0900
@@ -11,6 +11,9 @@
           fmttriggers=${fmtcomm},etex,ptex-base
 #
 execute addKanjiMap uptex-@kanjiEmbed@@kanjiVariant@.map
+execute addKanjiMap uptex-sc-@scEmbed@.map
+execute addKanjiMap uptex-tc-@tcEmbed@.map
+execute addKanjiMap uptex-ko-@koEmbed@.map
 
 binpattern f bin/${ARCH}/euptex
 binpattern f bin/${ARCH}/upbibtex
